src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+import java.util.Set;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(Storage storage) {
+        Map<String, Integer> fruitMap = storage.getAll();
+        Set<String> fruits = fruitMap.keySet();
+        StringBuilder sb = new StringBuilder();
+        for (String fruit : fruits) {
+            sb.append("\n").append(fruit).append(fruitMap.get(fruit));
+        }
+        return sb.substring(1);
```

Comment:
Using `sb.substring(1)` to remove the initial newline character is not the best approach. It's better to avoid adding the newline on the first iteration. You can check if the StringBuilder is empty or not before appending the newline character.